dags/tests/test_property_definitions.py

```diff
@@ -0,0 +1,236 @@
+import json
+import time
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from uuid import UUID
+
+import dagster
+
+from dags.property_definitions import (
+    DetectPropertyTypeExpression,
+    PropertyDefinitionsConfig,
+    property_definitions_ingestion_job,
+    setup_job,
+)
+from posthog.clickhouse.cluster import ClickhouseCluster, Query
+from posthog.models.property_definition import PropertyDefinition
+
+
+@dataclass
+class PropertyTypeTestData:
+    properties: str
+    expected: list[tuple[str, str | None]]
+
+
+def test_detect_property_type_expression(cluster: ClickhouseCluster) -> None:
```

Comment:
Would probably be good to refactor this into a parameterized test instead of one big mega test but that will require a bit of faffing with fixture setup